perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,112 @@
+		String url = "http://www.ga4gh.org/beacon/bonafide/ver1.0";
+
+		try {
+			Attribute statusAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF + ":elixirBonaFideStatusREMS");
+			String statusAttrValue = (String) statusAttr.getValue();
+			if (statusAttrValue != null && !statusAttrValue.isEmpty()) {
+				attribute.setValue(url);
+			} else {
+				Attribute userAttribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_VIRT + ":eduPersonScopedAffiliations");
+				if (userAttribute.getValue() != null) {
+					List<String> affiliations = (List<String>) userAttribute.getValue();
+					for (String affiliation : affiliations) {
+						if (affiliation != null && affiliation.startsWith("faculty@")) {
+							attribute.setValue(url);
+							break;
+						}
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read elixirBonaFideStatusREMS from user " + user.getId(), e);
+		}
+		return attribute;
+	}
+
+	@Override
+	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl sess, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<String> resolvingMessages = new ArrayList<>();
+		if (message == null) return resolvingMessages;
+
+		Matcher userSetMatcher = userSetPattern.matcher(message);
+		Matcher userRemoveMatcher = userRemovePattern.matcher(message);
+		Matcher userAllAttrsRemovedMatcher = userAllAttrsRemovedPattern.matcher(message);
+		Matcher uesAllAttrsRemovedMatcher = uesAllAttrsRemovedPattern.matcher(message);
+		Matcher uesSetAffiliationAttributeMatcher = uesSetAffiliationAttributePattern.matcher(message);
+		Matcher uesRemoveAffiliationAttributeMatcher = uesRemoveAffiliationAttributePattern.matcher(message);
+
+		User user;
+		Attribute attribute;
+
+		if (uesSetAffiliationAttributeMatcher.find() ||	uesRemoveAffiliationAttributeMatcher.find() ||	uesAllAttrsRemovedMatcher.find() ||
+				userSetMatcher.find() || userRemoveMatcher.find() || userAllAttrsRemovedMatcher.find()) {
+
+			user = sess.getPerunBl().getModulesUtilsBl().getUserFromMessage(sess, message);
+			if (user != null) {
+				String messageAttributeSet;
+
+				attribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_VIRT + ":elixirBonaFideStatus");
+				String value = (String) attribute.getValue();
+
+				if (value == null || value.isEmpty()) {
+					AttributeDefinition attributeDefinition = new AttributeDefinition(attribute);
+					messageAttributeSet = attributeDefinition.serializeToString() + " removed for " + user.serializeToString() + ".";
+				} else {
+					messageAttributeSet = attribute.serializeToString() + " set for " + user.serializeToString() + ".";
+				}
+				resolvingMessages.add(messageAttributeSet);
+			}
```

Comment:
If user is null, it is probably not totally OK. There should be at least log error about this situation.